Sources/StreamChat/Database/DatabaseContainer.swift

```diff
@@ -257,31 +239,20 @@ class DatabaseContainer: NSPersistentContainer {
         }
 
         writableContext.perform { [weak self] in
-            self?.sendNotificationForAllContexts(name: Self.WillRemoveAllDataNotification)
-
-            // Because this is a critical part of Stream, we don't want to make such changes unless it is in a controlled environment
-            if StreamRuntimeCheck._isBackgroundMappingEnabled {
-                // Before we remove the persistent store, we need to reset the state of the `NSManagedObjectContext`s to
-                // avoid a context from trying to access data from a previous store
-                self?.allContext.forEach { context in
-                    context.performAndWait { context.reset() }
+            let entityNames = self?.managedObjectModel.entities.compactMap(\.name)
+            var deleteError: Error?
+            entityNames?.forEach { [weak self] entityName in
+                let deleteFetch = NSFetchRequest<NSFetchRequestResult>(entityName: entityName)
+                let deleteRequest = NSBatchDeleteRequest(fetchRequest: deleteFetch)
+                do {
+                    try self?.writableContext.execute(deleteRequest)
+                    try self?.writableContext.save()
+                } catch {
+                    log.error("Batch delete request failed with error \(error)")
+                    deleteError = error
```

Comment:
Can we simulate one failure and inspect the db file that remains?

Reply:
maybe just put an if statement for a particular fetch request and skip the delete? Find the db file from the file system on a simulator, and let's inspect it.

Reply:
### Removing all data

Before:
<img width="906" alt="image" src="https://github.com/GetStream/stream-chat-swift/assets/12814114/ecf782fe-2999-496c-888d-d5d57b11c083">

After:
<img width="911" alt="image" src="https://github.com/GetStream/stream-chat-swift/assets/12814114/4063121e-7d85-4205-b66e-466de1dfcdff">

### Removing all data with mute fails
Before:
<img width="910" alt="image" src="https://github.com/GetStream/stream-chat-swift/assets/12814114/e0022ee3-b2f9-46bf-8dc9-3215afd1f5d1">

After:
<img width="902" alt="image" src="https://github.com/GetStream/stream-chat-swift/assets/12814114/68adbac5-37f3-435f-b569-a7ada4e99456">

We can see in both cases, there is a signifcant drop of MB's in the `8br4watad788` db file. But ofc, we can't really read it to check what is happening since the file is not readable.